src/main/java/v/hryhoryk/onlinebookstore/service/user/UserServiceImpl.java

```diff
@@ -26,10 +27,29 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         if (userRepository.existsByEmail(requestDto.email())) {
             throw new RegistrationException("Registration failed. User already exists.");
         }
-        Role roleByName = roleRepository.getRoleByName(Role.RoleName.ROLE_USER);
         User user = userMapper.toUser(requestDto);
+        createShoppingCart(user);
+        setUserRole(user);
+        setUserPassword(user, requestDto);
+        return saveUserAndReturnDto(user);
+    }
+
+    private void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        user.setShoppingCart(shoppingCart);
+    }
+
+    private void setUserRole(User user) {
+        Role roleByName = roleRepository.getRoleByName(Role.RoleName.ROLE_USER);
         user.setRoles(Set.of(roleByName));
+    }
+
+    private void setUserPassword(User user, UserRegistrationRequestDto requestDto) {
         user.setPassword(passwordEncoder.encode(requestDto.password()));
+    }
+
+    private UserResponseDto saveUserAndReturnDto(User user) {
```

Comment:
looks like this method is redundant, there is no logic that should be in private method